pkg/blockstorage/vmware/vmware.go

```diff
@@ -251,38 +256,43 @@ var reVslmSyncFaultFatal = regexp.MustCompile("Change tracking invalid or disk i
 // SnapshotCreate is part of blockstorage.Provider
 func (p *FcdProvider) SnapshotCreate(ctx context.Context, volume blockstorage.Volume, tags map[string]string) (*blockstorage.Snapshot, error) {
 	var res types.AnyType
+	description := generateSnapshotDescription(tags)
 	err := wait.PollImmediate(time.Second, defaultRetryLimit, func() (bool, error) {
-		log.Debug().Print("CreateSnapshot", field.M{"VolumeID": volume.ID})
-		task, lerr := p.Gom.CreateSnapshot(ctx, vimID(volume.ID), noDescription)
-		if lerr != nil {
-			return false, errors.Wrap(lerr, "CreateSnapshot task creation failure")
+		timeOfCreateSnapshotCall := time.Now()
```

Comment:
This should be set before the call to `wait.PollImmediate` function unless you want it reset within the implicit polling loop.

Reply:
I put it inside PollImmediate because we  will try to create a new snapshot on the next run, so there will be leakage on kanister level anyway. So we only re-try to get snapshot created in the current loop. The higher level logic  (that's why we adding a more reliable way to list snapshots)